src/lib/components/InfiniteScroll.svelte

```diff
@@ -0,0 +1,170 @@
+    const get_posts = async () => {
+        if (state === 'loading') return;
+        state = 'loading';
+        try {
+            console.log('get_posts', { sort, filter });
+            const search = new URLSearchParams({
+                page: `${page++}`,
+                type: 'home',
+                sort: sort.value as 'votes' | 'date',
+                filter: filter.value as 'all' | 'subscribed',
+                reverseSort: reverseSort + '', // default behaviour, but ts is ts
+            });
+            const res = await fetch(`${url}?${search}`);
+
+            if (res.status !== 200) {
+                throw new Error(await res.text());
+            }
+
+            const new_posts = await res.json();
+            buffer = buffer.concat(new_posts);
+            if (new_posts.length < PAGE_SIZE) {
+                state = 'no-posts';
+            } else {
+                state = 'active';
+            }
+            console.log('done loading');
+        } catch (e) {
+            toast.error('Unexpected error while loading posts');
+            console.error(e);
+            state = 'error';
+        }
+    };
+
+    const enter = () => {
+        get_posts();
+    };
+
+    const update = () => {
+        console.log('update');
+        buffer = [];
+        get_posts();
+    };
+
+    onMount(get_posts);
+</script>
+
+<div class="flex h-full max-h-full w-full flex-col items-center overflow-auto">
+    <div class="mx-4 mb-2 mt-4 flex flex-row gap-4 self-end">
+        <Select.Root
+            portal={null}
+            selected={filter}
+            onSelectedChange={(e) => {
+                filter = e;
+                update();
+            }}
+        >
+            <Select.Trigger class="w-[180px]">
+                <Select.Value placeholder="Filter" />
+            </Select.Trigger>
+            <Select.Content>
+                <Select.Item value="all" label="All">All</Select.Item>
+                <Select.Item value="subscribed" label="Subscribed" disabled={!data.signed_in}
+                    >Subscribed</Select.Item
+                >
+            </Select.Content>
+        </Select.Root>
+        <Select.Root
+            portal={null}
+            selected={sort}
+            onSelectedChange={(e) => {
+                sort = e;
+                update();
+            }}
+        >
+            <Select.Trigger class="w-[180px]">
+                <Select.Value placeholder="Sort" />
+            </Select.Trigger>
+            <Select.Content>
+                <Select.Item value="date" label="Date Posted">
+                    <Clock class="mr-2 h-4 w-4" />
+                    <span>Date Posted</span>
+                </Select.Item>
+                <Select.Item value="votes" label="Top">
+                    <ArrowUp class="mr-2 h-4 w-4" />
+                    <span>Top</span>
+                </Select.Item>
+            </Select.Content>
+        </Select.Root>
```

Comment:
I think there's a similar problem here where I think it'd likely be best for this to be handled by the parent component. In my soon-to-merge PR, we have a "Popular" route in the navbar which presumably would show the user the most popular posts regardless of whether they're subscribed. This would mean there exist pages where at least one of these select options isn't necessary.